### PR TITLE
Implement spi layer for Logging service

### PR DESCRIPTION
**Pull request by mziccard**
System tests will come later with `LoggingImpl`.

### CODE REVIEW

File: gcloud-java-logging/src/main/java/com/google/cloud/logging/spi/DefaultLoggingRpc.java [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+  private static final class InternalLoggingOptions extends LoggingOptions {
+
+    private static final long serialVersionUID = -2107638980310672033L;
+
+    private InternalLoggingOptions(LoggingOptions options) {
+      super(options.toBuilder());
+    }
+
+    @Override
+    protected ExecutorFactory executorFactory() {
```
Comment (mziccard) [inline gcloud-java-logging/src/main/java/com/google/cloud/logging/spi/DefaultLoggingRpc.java]: Codacy Issue found: [Overriding method merely calls super](https://www.codacy.com/app/mziccard/gcloud-java/file/2755575837/issues/source?bid=3123104&fileBranchId=3358807#l92)

File: gcloud-java-logging/src/main/java/com/google/cloud/logging/LoggingOptions.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      return new LoggingOptions(this);
+    }
+  }
+
+  protected LoggingOptions(Builder builder) {
+    super(LoggingFactory.class, LoggingRpcFactory.class, builder);
+  }
+
+  @Override
+  protected ExecutorFactory executorFactory() {
```
Comment (mziccard) [inline gcloud-java-logging/src/main/java/com/google/cloud/logging/LoggingOptions.java]: Codacy Issue found: [Overriding method merely calls super](https://www.codacy.com/app/mziccard/gcloud-java/file/2755575836/issues/source?bid=3123104&fileBranchId=3358807#l91)

**Comment by lesv** [comment]
LGTM - though you might wish to look at codecy/pr's comments and fix them -- or comment why we need redundant methods.

**Comment by mziccard** [comment]
@lesv all overrides claimed to be redundant by codacy in this PR are made so that overridden methods are visible in the "logging.spi" package.

**Pull request merged**